roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-domain.conf.j2

```diff
@@ -129,6 +129,20 @@
 	}
 	{% endif %}
 
+    {% if matrix_ldap_registration_proxy_enabled %}
+        location _matrix/client/r0/register {
+            {% if matrix_nginx_proxy_enabled %}
+                {# Use the embedded DNS resolver in Docker containers to discover the service #}
+                resolver 127.0.0.11 valid=5s;
+                set $backend "{{ matrix_nginx_proxy_proxy_matrix_3pid_registration_addr_with_container }}";
+                proxy_pass http://$backend/register;
+            {% else %}
+                {# Generic configuration for use outside of our container setup #}
+                proxy_pass http://{{ matrix_nginx_proxy_proxy_matrix_3pid_registration_addr_sans_container }}/register;
+            {% endif %}
+        }
+    {% endif %}
```

Comment:
There are 2 ways to do this in a more independent way (one that doesn't cross-reference variables between `matrix-nginx-proxy` and the `matrix-ldap-registration-proxy` role:

One is by introducing new variables in the `matrix-nginx-proxy` role:
- for enabling this behavior (disabled by default)
- for the actual addresses that it will proxy to

`matrix_nginx_proxy_proxy_matrix_3pid_registration_addr_with_container` and `matrix_nginx_proxy_proxy_matrix_3pid_registration_addr_sans_container` that you're using here are used for other purposes (forwarding email 3pid stuff to ma1sd).. Not sure if it's a good idea to reuse them for your own purposes - for proxying to `ldap-registration-proxy`. I also don't see any logic that points these variables to `ldap-registration-proxy`, so I'm not sure how it's expected to work.. Or am I missing something?

-----

The other (**better**) way to solve this is **without adding new variables** to the `matrix-nginx-proxy` role.

Your role can inject additional configuration into the `matrix-nginx-proxy` role by injecting blocks into `matrix_nginx_proxy_proxy_matrix_additional_server_configuration_blocks` at runtime.

This is typically done from the role's `tasks/init.yml` file.

See an example from the `matrix-bridge-mautrix-telegram` role here https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/427806674980325d308987661df305d7ba694b02/roles/matrix-bridge-mautrix-telegram/tasks/init.yml#L30-L75

For this to work, your role needs to be defined **before** the `matrix-nginx-proxy` role in the main [`setup.yml` file](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/setup.yml). Speaking of which, I don't see you adjusting the `setup.yml` file in this PR, so this role will not be taken into account at all when running the playbook.

Reply:
Done